crates/sargon/src/system/sargon_os/transactions/support/signable_manifest_summary.rs

```diff
@@ -0,0 +1,151 @@
+use crate::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SignableManifestSummary {
+    pub id: Exactly32Bytes,
+    pub summary: ManifestSummary,
+}
+
+impl SignableManifestSummary {
+    pub fn new(summary: ManifestSummary) -> Self {
+        Self {
+            id: Exactly32Bytes::generate(),
+            summary: summary,
+        }
+    }
+}
+
+impl HasSampleValues for SignableManifestSummary {
+    fn sample() -> Self {
+        Self {
+            id: Exactly32Bytes::sample(),
+            summary: ManifestSummary::sample(),
+        }
+    }
+
+    fn sample_other() -> Self {
+        Self {
+            id: Exactly32Bytes::sample_other(),
+            summary: ManifestSummary::sample_other(),
+        }
+    }
+}
+
+impl SignableID for Exactly32Bytes {}
+
+impl Into<Exactly32Bytes> for SignableManifestSummary {
```

Comment:
prefer `impl From`, **always**, so `impl From<SignableManifestSummary> for Exactly32Bytes`  (Ghenadie has used `Into` imside the `sargon-uniffi` crate because he cannot impl `From<InternalType>`  due to orphan rule)